pallets/stream-payment/src/migrations.rs

```diff
@@ -0,0 +1,145 @@
+    pub asset_id: AssetId,
+    /// Amount of asset / unit.
+    pub rate: BalanceOrDuration,
+}
+
+#[derive(Encode, Decode, Clone)]
+pub struct OldChangeRequest<Unit, AssetId, Balance> {
+    pub requester: crate::Party,
+    pub kind: crate::ChangeKind<Balance>,
+    pub new_config: OldStreamConfig<Unit, AssetId, Balance>,
+    pub deposit_change: Option<crate::DepositChange<Balance>>,
+}
+
+pub struct MigrateStreamPaymentNewConfigFields<T>(pub PhantomData<T>);
+impl<T> Migration for MigrateStreamPaymentNewConfigFields<T>
+where
+    T: Config,
+{
+    fn friendly_name(&self) -> &str {
+        "TM_MigrateStreamPaymentNewConfigFields"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        let mut count = 0;
+        crate::Streams::<T>::translate(|_key, value: OldStreamOf<T>| {
+            count += 1;
+            let OldStream {
+                source,
+                target,
+                deposit,
+                last_time_updated,
+                request_nonce,
+                pending_request,
+                opening_deposit,
+                config:
+                    OldStreamConfig {
+                        time_unit,
+                        asset_id,
+                        rate,
+                    },
+            } = value;
+
+            let pending_request = pending_request.map(
+                |OldChangeRequest {
+                     requester,
+                     kind,
+                     new_config:
+                         OldStreamConfig {
+                             time_unit,
+                             asset_id,
+                             rate,
+                         },
+                     deposit_change,
+                 }| crate::ChangeRequest {
+                    requester,
+                    kind,
+                    deposit_change,
+                    new_config: crate::StreamConfig {
+                        time_unit,
+                        asset_id,
+                        rate,
+                        minimum_request_deadline_delay: 0u32.into(),
+                        soft_minimum_deposit: 0u32.into(),
+                    },
+                },
+            );
+
+            Some(crate::Stream {
+                source,
+                target,
+                deposit,
+                last_time_updated,
+                request_nonce,
+                pending_request,
+                opening_deposit,
+                config: crate::StreamConfig {
+                    time_unit,
+                    asset_id,
+                    rate,
+                    minimum_request_deadline_delay: 0u32.into(),
+                    soft_minimum_deposit: 0u32.into(),
+                },
+            })
+        });
+
+        let db_weights = T::DbWeight::get();
+        db_weights.reads_writes(count, count)
+    }
+
+    #[cfg(feature = "try-runtime")]
+    fn pre_upgrade(&self) -> Result<Vec<u8>, sp_runtime::DispatchError> {
+        Ok(vec![])
```

Comment:
I'd say let's write pre-upgrade and post upgrade hooks. I think they should be easily testable with try-runtime.

Essentially, read one (if any) of the streams before, return it encoded. Receive it in post-upgrade encoded, decode it, and construct the new struct with the expected new fields. Verify that the storage has been migrated accordingly